components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/CORSRequestHandler.java

```diff
@@ -79,7 +80,14 @@ void initializeHeaders() {
 		if (allowedMethods == null) {
 			allowedMethods = APIUtil.getAllowedMethods();
 		}
-
+		if (accessControlMaxAge != null && accessControlMaxAge.trim().length() == 0) {
```

Comment:
`accessControlMaxAge.trim().length() == 0` should be `accessControlMaxAge.trim().length() != 0`